File: applications/app-bemyplan/src/main/java/com/deploy/bemyplan/user/service/dto/request/CheckAvailableNameRequestDto.java
package com.deploy.bemyplan.user.service.dto.request;


import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Pattern;

@ToString
@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class CheckAvailableNameRequestDto {

    @NotBlank(message = "{user.nickname.notBlank}")
    @Pattern(regexp = "^[ㄱ-ㅎㅏ-ㅣ가-힣a-zA-Z0-9._]{1,15}$", message = "{user.nickname.format}")
    private String nickname;
}

File: applications/app-bemyplan/src/main/kotlin/com/bemyplan/auth/application/CheckAvailableNameRequestDto.kt
package com.bemyplan.auth.application

import javax.validation.constraints.NotBlank
import javax.validation.constraints.Pattern

data class CheckAvailableNameRequestDto(
    @NotBlank(message = "{user.nickname.notBlank}")
    @Pattern(regexp = "^[ㄱ-ㅎㅏ-ㅣ가-힣a-zA-Z0-9._]{1,15}$", message = "{user.nickname.format}")
    val nickname: String,
)
